arangod/Replication2/ReplicatedLog/LogFollower.cpp

```diff
@@ -140,19 +148,26 @@ auto replicated_log::LogFollower::appendEntries(AppendEntriesRequest req)
   MeasureTimeGuard measureTimeGuard{
       _logMetrics->replicatedLogFollowerAppendEntriesRtUs};
 
-  auto self = _guardedFollowerData.getLockedGuard();
+  auto dataGuard = _guardedFollowerData.getLockedGuard(ADB_HERE);
 
   {
     // Preflight checks - does the leader, log and other stuff match?
     // This code block should not modify the local state, only check values.
-    if (auto result = appendEntriesPreFlightChecks(self.get(), req);
+    if (auto result = appendEntriesPreFlightChecks(dataGuard.get(), req);
         result.has_value()) {
       return *result;
     }
 
-    self->_lastRecvMessageId = req.messageId;
+    dataGuard->_lastRecvMessageId = req.messageId;
+    _appendEntriesInFlight = true;
   }
 
+  // In case of an exception, this scope guard sets the in flight flag to false.
+  // _appendEntriesInFlight is an atomic variable, hence we are allowed to set
+  // it without acquiring the mutex.
+  auto inFlightScopeGuard =
+      ScopeGuard([&flag = _appendEntriesInFlight]() noexcept { flag = false; });
```

Comment:
Alternatively, you could acquire the lock inside the ScopeGuard; then in `checkResultAndCommitIndex` acquire the lock, cancel the `inFlightScopeGuard` afterwards, and then set the flag to false.